test/search.cc

```diff
@@ -177,12 +183,18 @@ void search(const valhalla::baldr::Location& location,
     throw std::logic_error("Got more edges than expected");
 }
 
-void search(const valhalla::baldr::Location& location, size_t result_count, int reachability) {
+void search(valhalla::baldr::Location location, size_t result_count, int reachability) {
+
   // make the config file
   boost::property_tree::ptree conf;
   conf.put("tile_dir", tile_dir);
-
   valhalla::baldr::GraphReader reader(conf);
+
+  // send it to pbf and back just in case something is wrong with that conversion
+  valhalla::odin::Location pbf;
+  PathLocation::toPBF(location, &pbf, reader);
+  location = PathLocation::fromPBF(pbf);
```

Comment:
these changes are important, they force the loki tests to convert between custom classes and pbf to ensure that both the services (via json->pbf) and direct use of the api (custom classes) will work.